childimage/NSIB: Remove non-functional environmental key support (#17088)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

  .. note::
     This has security implications and may allow secrets to be leaked to the non-secure application in RAM.

> * Removed non-working support for setting the NSIB signing key using either environmental or command line variable ``SB_SIGNING_KEY_FILE`` along with child_image.

Reviewer: ```suggestion
* Removed the non-working support for configuring the NSIB signing key through the environmental or command line variable (``SB_SIGNING_KEY_FILE``), along with child image.
```

>   ..note::
>     This never been working. Use any Kconfig method for configuring the signing key.

Reviewer: ```suggestion
  ..note::
    This feature has never been functional. 
    To configure the signing key, use any available Kconfig method.
```
